integrations/influxdb_metrics_dashboard/lambda/upload_dashboard/main.go

```diff
@@ -0,0 +1,459 @@
+		}
+	}
+
+	if waiterInterval == MaxWaitIntervals {
+		return "", fmt.Errorf("Failed to create service account due to workspace creation timeout")
+	}
+
+	serviceAccountOutput, err := grafanaClient.CreateWorkspaceServiceAccount(
+		context.TODO(),
+		&grafana.CreateWorkspaceServiceAccountInput{
+			GrafanaRole: grafanaTypes.RoleAdmin,
+			Name:        &workspaceName,
+			WorkspaceId: grafanaWorkspace.Id,
+		},
+	)
+	if err != nil {
+		log.Printf("Error creating workspace service account: %s", err)
+		return "", err
+	}
+	serviceAccountTokenOutput, err := grafanaClient.CreateWorkspaceServiceAccountToken(context.TODO(), &grafana.CreateWorkspaceServiceAccountTokenInput{
+		Name:             &grafanaServiceAccountTokenName,
+		SecondsToLive:    &grafanaServiceAccountTokenSecondsToLive,
+		ServiceAccountId: serviceAccountOutput.Id,
+		WorkspaceId:      grafanaWorkspace.Id,
+	})
+	if err != nil {
+		log.Printf("Error getting Grafana service account token: %s", err)
+		return "", err
+	}
+	if serviceAccountTokenOutput.ServiceAccountToken.Key == nil {
+		log.Printf("Error getting Grafana service account token: %s", err)
+		return "", fmt.Errorf("Failed to get token key")
+	}
+	serviceAccountTokenKey = *serviceAccountTokenOutput.ServiceAccountToken.Key
+
+	// The endpoint may take time to populate from the workspace output
+	waiterInterval = 0
+	for waiterInterval < MaxWaitIntervals {
+		grafanaWorkspace, err = getWorkspaceByName(grafanaClient, workspaceName)
+		if err != nil {
+			log.Printf("Failed to get workspace: %s", err)
+			return "", err
+		}
+		if grafanaWorkspace.Endpoint != nil {
+			break
+		}
+		waiterInterval++
+		time.Sleep(SleepDuration * time.Second)
+	}
+
+	ret, err := uploadDashboard(serviceAccountTokenKey, *grafanaWorkspace.Endpoint, datasourceName, dashboardName, databaseName)
+	if err != nil {
+		return "", err
+	}
+
+	return ret, nil
+}
+
+func lambdaHandler(ctx context.Context, event map[string]interface{}) (events.APIGatewayProxyResponse, error) {
+	resp, err := createGrafanaDashboard()
+	if err != nil {
+		log.Printf("Error creating Grafana dashboard: %v", err)
+		return events.APIGatewayProxyResponse{
+			StatusCode: http.StatusInternalServerError,
+			Body:       fmt.Sprintf("Error creating Grafana dashboard: %v", err),
+		}, nil
+	}
+
+	return events.APIGatewayProxyResponse{
+		StatusCode: http.StatusOK,
+		Body:       resp,
+	}, nil
+}
+
+func main() {
+	lambda.Start(lambdaHandler)
+}
+
+type panelField struct {
+	gridPosition map[string]interface{}
+	title        string
+	panelType    string
+	refId        string
+	query        string
+}
+
+func generatePanels(datasourceName string, databaseName string) []interface{} {
+
+	panelFields := []panelField{
+		{map[string]interface{}{"h": 6, "w": 7, "x": 0, "y": 0}, "Bucket Cardinality", "stat", "A", fmt.Sprintf("SELECT time,gauge FROM \"%s\".\"storage_bucket_series_num\" WHERE time between ago(15m) and now() group by 1,2 ORDER BY time,gauge DESC LIMIT 1", databaseName)},
```

Comment:
If you put the time range as a default value in the dashboard and not the query for the panel, then users can change the time range. I believe putting the time range in the panel query means the time range cannot be changed by users.

I believe you can do this by, in `generateDashboard`, adding this:
```
	"dashboard": map[string]interface{}{
			"panels": generatePanels(datasourceName, databaseName),
			"title":  dashboardName,
            "time": map[string]interface{}{
                "from": "now-15m",
                "to": "now",
            },
		},
```